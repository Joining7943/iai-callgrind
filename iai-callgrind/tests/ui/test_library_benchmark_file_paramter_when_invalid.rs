use iai_callgrind::library_benchmark;

#[library_benchmark]
#[benches::my_id(file = "iai-callgrind/tests/fixtures/empty.fix")]
fn bench10(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

#[library_benchmark]
#[benches::my_id(file = "iai-callgrind/tests/fixtures/does_not_exist")]
fn bench20(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

#[library_benchmark]
#[benches::my_id(file = "iai-callgrind/tests/fixtures/invalid-utf8.fix")]
fn bench30(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

// both parameters are valid but specifying both is invalid
#[library_benchmark]
#[benches::my_id(file = "iai-callgrind/tests/fixtures/numbers.fix", args = [("valid_arg".to_owned()), "another".to_owned()])]
fn bench40(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

#[library_benchmark]
#[benches::my_id(file = "iai-callgrind/tests/fixtures/numbers.fix")]
fn bench50(value: u64) -> String {
    value.to_string()
}

#[library_benchmark]
#[benches::my_id(file = ("iai-callgrind/tests/fixtures/numbers.fix", String))]
fn bench60(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

#[library_benchmark]
#[benches::my_id(file = String::from("iai-callgrind/tests/fixtures/numbers.fix"))]
fn bench70(value: String) -> u64 {
    value.parse::<u64>().unwrap()
}

fn main() {}
