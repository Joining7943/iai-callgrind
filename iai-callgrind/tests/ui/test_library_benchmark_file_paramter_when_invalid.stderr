error: The provided file 'iai-callgrind/tests/fixtures/empty.fix' was empty
 --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:4:25
  |
4 | #[benches::my_id(file = "iai-callgrind/tests/fixtures/empty.fix")]
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Error opening '$WORKSPACE/target/tests/trybuild/iai-callgrind/iai-callgrind/tests/fixtures/does_not_exist': No such file or directory (os error 2)
  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:10:25
   |
10 | #[benches::my_id(file = "iai-callgrind/tests/fixtures/does_not_exist")]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Error reading line 1 in file '$WORKSPACE/target/tests/trybuild/iai-callgrind/iai-callgrind/tests/fixtures/invalid-utf8.fix': stream did not contain valid UTF-8
  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:16:25
   |
16 | #[benches::my_id(file = "iai-callgrind/tests/fixtures/invalid-utf8.fix")]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Only one parameter of `file` or `args` can be present
  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:23:25
   |
23 | #[benches::my_id(file = "iai-callgrind/tests/fixtures/numbers.fix", args = [("valid_arg".to_owned()), "another".to_owned()])]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Invalid value for `file`

         = help: The `file` argument needs a literal string containing the path to an existing file at compile time
         = note: `file = "benches/some_fixture"`

  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:35:25
   |
35 | #[benches::my_id(file = ("iai-callgrind/tests/fixtures/numbers.fix", String))]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Invalid value for `file`

         = help: The `file` argument needs a literal string containing the path to an existing file at compile time
         = note: `file = "benches/some_fixture"`

  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:41:25
   |
41 | #[benches::my_id(file = String::from("iai-callgrind/tests/fixtures/numbers.fix"))]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> tests/ui/test_library_benchmark_file_paramter_when_invalid.rs:29:25
   |
29 | #[benches::my_id(file = "iai-callgrind/tests/fixtures/numbers.fix")]
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                         |
   |                         expected `u64`, found struct `String`
   |                         arguments to this function are incorrect
   |
note: function defined here
  --> $RUST/core/src/hint.rs
   |
   | pub const fn black_box<T>(dummy: T) -> T {
   |              ^^^^^^^^^
